src/components/panels/EnterpriseListPanelComponent.js

```diff
@@ -0,0 +1,89 @@
+'use strict';
+
+import React from 'react';
+
+require('styles/panels/EnterpriseListPanel.scss');
+
+class EnterpriseListPanelComponent extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      enterprises: props.enterprises,
+      status: props.status
+    };
+  }
+
+  getEnterpriseDetails(enterpriseId) {
+    const api_root = this.context.config.api_root;
+
+    let locales = this.context.config.locales;
+
+    // Fetch enterprise details in all locales
+    // Note: we could defer this until the user clicks on a locale tab
+    //       if fetching all locales takes a long time
```

Comment:
Another option is there is also an endpoint:
/enterprise/{id}/complete

That returns all the enterprise info in one response